packages/serverpod/lib/src/server/endpoint_dispatch.dart

```diff
@@ -336,13 +359,102 @@ class MethodStreamCallContext {
   /// The input streams to pass to the method.
   final List<StreamParameterDescription> inputStreams;
 
+  OnRevokedAuthenticationCallback? _onRevokedAuthentication;
+
+  MessageCentralListenerCallback? _revokedAuthenticationCallback;
+
+  final Session _session;
+
+  AuthenticationInfo? _authenticationInfo;
+
   /// Creates a new [MethodStreamCallContext].
   MethodStreamCallContext({
     required this.method,
     required this.arguments,
     required this.inputStreams,
     required this.endpoint,
-  });
+    required Session session,
+  }) : _session = session;
+
+  /// Initializes the context.
+  Future<void> init() async {
```

Comment:
Suggestion/Discussion: I'm not a big fan of init methods. Is there no way to have this as part of a factory or static factory method instead?


So that once the MethodStreamCallContext is returned it is already initialied?